### PR TITLE
024: Add Remaining Requirements after scanning transcript

### PR DESCRIPTION
**Pull request by calvinqc**
1. Move all helper functions to utils.js to avoid importing the whole controller
2. Improve Transcript Scanning to parse all courses that are in the database.
3. Complete Get Remaining Requirements after scanning the transcript.
4. Fix bugs when scanning for transcript
5. Adding userID for each plan (if some users have the same planID, and one of them updates the plan, then it will update for all other users. This is not good. Hence, each user will have a unique degree Plan)
6. Add 3 files:
   - SE-program: all requirements (GE, major, technical elective)
   - CloudOCRResponse: the response after handling the OCR Result
   - ScanResult: Result after Scanning the transcript
7. Remove key and uploads/ folder

### CODE REVIEW

File: server/package.json [hunk 1]
```diff
@@ -3,8 +3,8 @@
   "version": "1.0.0",
   "private": true,
   "scripts": {
-    "start": "source .env && ENV=prod nodemon -r esm app.js",
-    "dev": "source .env && ENV=dev nodemon -r esm app.js",
+    "start": "ENV=prod node -r esm app.js",
```
Comment (DaleCS) [inline server/package.json:6]: "start" script no longer uses nodemon. Is this correct?

**Review by DaleCS — CHANGES_REQUESTED** [review]
The server/package.json "start" script no longer uses nodemon. Is this correct?

Do you have a test .pdf file that I can use to test this with? I used my own transcript and it showed that I didn't have any remain courses left which is expected but I want to see what the getRemainingRequirement() function outputs with a transcript that still has remaining requirements.

Other than those, the code looks good to me.

File: server/package.json (hunk 1 above)
Comment (calvinqc) [reply server/package.json:6]: Yes, this is for the purpose when deploy to Gcloud App Engine

**Review by DaleCS — APPROVED** [review]
LGTM

**Comment by ellamaolson** [comment]
LGTM

**Review by ellamaolson — APPROVED** [review]

**Pull request merged**